src/main/scala/beam/agentsim/agents/ridehail/RideHailManager.scala

```diff
@@ -297,8 +297,15 @@ class RideHailManager(
   private val initialNumHouseholdVehicles = scenario.getHouseholds.getHouseholds
     .values()
     .asScala
-    .map(_.getVehicleIds.size())
-    .sum / beamServices.beamConfig.beam.agentsim.agents.vehicles.fractionOfInitialVehicleFleet // Undo sampling to estimate initial number
+    .flatMap { hh =>
+      hh.getVehicleIds.asScala.map { vehId =>
```

Comment:
Just for future note - this seems like an area where `collect` would have been more apt. Basically anytime you see a map-filter - then think collect instead.